lib/run_printer.py

```diff
@@ -0,0 +1,87 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+
+"""Signal the run-build process to dump its run file and print it"""
+
+
+import dataclasses
+import json
+import os
+import pathlib
+import signal
+import sys
+import time
+
+import lib.litani
+import lib.litani_report
+import lib.pid_file
+
+
+"""When run-build receives this Unix signal, it will write the run file"""
+DUMP_SIGNAL = signal.SIGUSR1
+_DUMPED_RUN = "dumped-run.json"
+
+
+def _exit_success(run):
+    print(json.dumps(run, indent=2))
+    sys.exit(0)
+
+
+def _exit_error():
+    print(json.dumps(None))
+    sys.exit(0)
```

Comment:
Should this be `sys.exit(1)` ?

Reply:
Litani doesn't ever exit with non-zero unless it crashed (or unless the user asks it to, `litani run-build` has a flag to do that but it's not the default). This is because we usually have an in-band way of indicating failure. In this case, printing `null` is sufficient to indicate an error, because `null` is not a valid format for run.json. This is also made clear in the man page.